rclpy/rclpy/time_source.py

```diff
@@ -91,12 +97,14 @@ def attach_node(self, node):
 
     def detach_node(self):
         # Remove the subscription to the clock topic.
+        error = RuntimeError('Unable to destroy previously created clock subscription')
         if self._clock_sub is not None:
-            if self._node is None:
-                raise RuntimeError('Unable to destroy previously created clock subscription')
-            self._node.destroy_subscription(self._clock_sub)
+            node = self._get_node()
+            if node is None:
+                raise error
+            node.destroy_subscription(self._clock_sub)
         self._clock_sub = None
-        self._node = None
+        node = None
```

Comment:
Why is this line necessary? The local var should go out of scope at the end of the function (which is right after this line) anyway.

Reply:
Is this meant to be `self._node_weak_ref = None`? It seems like the weak reference isn't needed after `detach_node()` is called.

Reply:
Yes, @sloretz is correct.